ScanCommandExecutorTest is failing due to "scanner" is null"

## Issue

The java test `ScanCommandExecutorTest` is failing due to `"scanner" is null`

```
[INFO] Running dev.snowdrop.service.scanner.ScanCommandExecutorTest
2025-12-16 14:01:21,633 INFO  [dev.snowdrop.service.scanner.ScanCommandExecutor] (main) Scanner null found 2 matches
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.085 s <<< FAILURE! -- in dev.snowdrop.service.scanner.ScanCommandExecutorTest
[ERROR] dev.snowdrop.service.scanner.ScanCommandExecutorTest.executeCommandForQuery_returnsEmptyListWhenNoScannerFound -- Time elapsed: 0.077 s <<< ERROR!
java.lang.NullPointerException: Cannot invoke "dev.snowdrop.service.scanner.QueryScanner.scansCodeFor(dev.snowdrop.analyze.Config, dev.snowdrop.parser.Query)" because "scanner" is null
        at dev.snowdrop.service.scanner.ScanCommandExecutor.executeCommandForQuery(ScanCommandExecutor.java:41)
        at dev.snowdrop.service.scanner.ScanCommandExecutorTest.executeCommandForQuery_returnsEmptyListWhenNoScannerFound(ScanCommandExecutorTest.java:60)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

--- a/migration-cli/src/main/java/dev/snowdrop/mtool/commands/AnalyzeCommand.java
+++ b/migration-cli/src/main/java/dev/snowdrop/mtool/commands/AnalyzeCommand.java
@@ -57,7 +57,7 @@ public class AnalyzeCommand implements Runnable {
 	private String output;
 
 	@CommandLine.Option(names = {
-			"--scanner"}, description = "Scanner tool to be used to analyse the code: jdtls, openrewrite", defaultValue = "jdtls")
+			"--scanner"}, description = "Scanner tool to be used to analyse the code: jdtls, openrewrite", defaultValue = "openrewrite")
 	public String scanner;
 
 	@Override

--- a/scanner/src/main/java/dev/snowdrop/mtool/scanner/ScanCommandExecutor.java
+++ b/scanner/src/main/java/dev/snowdrop/mtool/scanner/ScanCommandExecutor.java
@@ -5,6 +5,7 @@ import dev.snowdrop.mtool.model.analyze.Match;
 import dev.snowdrop.mtool.model.parser.Query;
 import org.jboss.logging.Logger;
 
+import java.util.Collections;
 import java.util.List;
 
 public class ScanCommandExecutor {
@@ -23,22 +24,12 @@ public class ScanCommandExecutor {
 	}
 
 	public List<Match> executeCommandForQuery(Config config, Query query) {
-		/*
-		QueryScanner scanner = spiRegistry.findScannerFor(query);;
-		if (config.scanner() != null) {
-			Optional<QueryScanner> optScanner = spiRegistry.findScanner(config.scanner());
-			if (optScanner.isPresent() && optScanner.get().supports(query)) {
-				scanner = optScanner.get();
-				logger.debugf("Using command configured scanner %s", scanner.getScannerType());
-			} else {
-				logger.infof("Using %s as scanner for analysing code for query %s.%s", config.scanner(),
-						query.fileType(), query.symbol());
-				return Collections.emptyList();
-			}
-		}*/
+		QueryScanner scanner = spiRegistry.resolveScannerForQuery(config, query);
+		if (scanner == null) {
+			return Collections.emptyList();
+		}
 
-		QueryScanner scanner = spiRegistry.resolveScannerForQuery(query, config);
-		List<Match> scannerResults = scanner.scansCodeFor(config, query);
+		final List<Match> scannerResults = scanner.scansCodeFor(config, query);
 		logger.infof("Scanner %s found %d matches", scanner.getScannerType(), scannerResults.size());
 		return scannerResults;
 	}

--- a/scanner/src/main/java/dev/snowdrop/mtool/scanner/ScannerSpiRegistry.java
+++ b/scanner/src/main/java/dev/snowdrop/mtool/scanner/ScannerSpiRegistry.java
@@ -1,6 +1,7 @@
 package dev.snowdrop.mtool.scanner;
 
 import dev.snowdrop.mtool.model.analyze.Config;
+import dev.snowdrop.mtool.model.analyze.ScannerType;
 import dev.snowdrop.mtool.model.parser.Query;
 import org.jboss.logging.Logger;
 
@@ -28,12 +29,6 @@ public class ScannerSpiRegistry {
 		}
 	}
 
-	public QueryScanner findScannerFor(Query query) {
-		return scanners.values().stream().filter(queryScanner -> queryScanner.supports(query)).findFirst()
-				.orElseThrow(() -> new IllegalArgumentException(
-						"No scanner supports query: " + query.fileType() + "." + query.symbol()));
-	}
-
 	public Optional<QueryScanner> findScanner(String name) {
 		return Optional.ofNullable(scanners.get(name));
 	}
@@ -47,12 +42,13 @@ public class ScannerSpiRegistry {
 	 * 1. The default scanner (config.scanner) if it has been defined and if it supports the query.
 	 * 2. Any other scanner supporting the query (excluding the default one).
 	 *
-	 * @param query The query (fileType and symbol) to check support for.
 	 * @param config The application configuration containing the default scanner name.
+	 * @param query  The query (fileType and symbol) to check support for.
 	 * @return The resolved QueryScanner.
 	 * @throws IllegalArgumentException if no scanner can be found that supports the query.
 	 */
-	public QueryScanner resolveScannerForQuery(Query query, Config config) {
+	public QueryScanner resolveScannerForQuery(Config config, Query query) {
+		String fallbackScannerName = ScannerType.OPENREWRITE.label();
 		String configuredScannerName = config.scanner();
 
 		// --- Priority 1: Scanner defined as config parameter
@@ -81,6 +77,7 @@ public class ScannerSpiRegistry {
 					query.fileType(), query.symbol());
 			return resolvedScanner;
 		}
+		fallbackScanner = findScanner(fallbackScannerName);
 
 		return fallbackScanner.orElseThrow(() -> new IllegalArgumentException("No scanner supports query: "
 				+ query.fileType() + "." + query.symbol()

--- a/scanner/src/test/java/dev/snowdrop/mtool/scanner/ScanCommandExecutorTest.java
+++ b/scanner/src/test/java/dev/snowdrop/mtool/scanner/ScanCommandExecutorTest.java
@@ -3,7 +3,6 @@ package dev.snowdrop.mtool.scanner;
 import dev.snowdrop.mtool.model.analyze.Config;
 import dev.snowdrop.mtool.model.analyze.Match;
 import dev.snowdrop.mtool.model.parser.Query;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
@@ -18,7 +17,7 @@ import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
-@Disabled
+//@Disabled
 class ScanCommandExecutorTest {
 
 	@Mock
@@ -40,7 +39,7 @@ class ScanCommandExecutorTest {
 		List<Match> expectedResults = List.of(new Match("file1", "openrewrite", Collections.emptyList()),
 				new Match("file2", "maven", Collections.emptyList()));
 
-		when(spiRegistry.findScannerFor(query)).thenReturn(scanner);
+		when(spiRegistry.resolveScannerForQuery(config, query)).thenReturn(scanner);
 		when(scanner.scansCodeFor(config, query)).thenReturn(expectedResults);
 
 		// When
@@ -49,15 +48,13 @@ class ScanCommandExecutorTest {
 		// Then
 		assertEquals(expectedResults, result);
 		verify(scanner).scansCodeFor(config, query);
-		verify(spiRegistry).findScannerFor(query);
+		verify(spiRegistry).resolveScannerForQuery(config, query);
 	}
 
 	@Test
 	void executeCommandForQuery_returnsEmptyListWhenNoScannerFound() {
 		ScanCommandExecutor executor = new ScanCommandExecutor(spiRegistry);
 		Query query = new Query("java", "interface", Collections.emptyMap());
-
-		when(spiRegistry.findScannerFor(query)).thenReturn(null);
 
 		List<Match> result = executor.executeCommandForQuery(config, query);
 

